routing/osrm_path_segment_factory.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "routing/loaded_path_segment.hpp"
+
+#include "std/unique_ptr.hpp"
+
+namespace routing
+{
+struct RoutingMapping;
+struct RawPathData;
+struct FeatureGraphNode;
+
+namespace turns
+{
+using TSeg = OsrmMappingTypes::FtSeg;
+
+// General constructor.
+unique_ptr<LoadedPathSegment> LoadedPathSegmentFactory(RoutingMapping & mapping,
```

Comment:
I mean, LoadedPathSegment is not an interface and doesn't have deriving classes. So, unique_ptr<LoadedPathSegment> looks like an overkill.